spectator-reg-atlas/src/main/java/com/netflix/spectator/atlas/AtlasMeter.java

```diff
@@ -38,12 +39,19 @@ abstract class AtlasMeter implements Meter {
   /** Last time this meter was updated. */
   private volatile long lastUpdated;
 
+  /**
+   * Reference count for batch updaters. The meter cannot be expired while a batch updater
+   * instance is around.
+   */
+  private AtomicInteger batchUpdaterRefCount;
+
   /** Create a new instance. */
   AtlasMeter(Id id, Clock clock, long ttl) {
     this.id = id;
     this.clock = clock;
     this.ttl = ttl;
     lastUpdated = clock.wallTime();
+    batchUpdaterRefCount = new AtomicInteger(0);
```

Comment:
Is it worth looking at lazy initialization to avoid the overhead for every meter? Just worried about the high cardinality use case.

Reply:
I refactored it to avoid the ref count. It is a bit less self-contained as the swap types need to inspect updater and pass in a supplier so the meter can be resolved on updates in case the registry has expired them.